### DISCUSSION TITLE
Does fpdf2 generate PDF compliant with PDF/X-1a ?

### DISCUSSION THREAD

**Discussion by dgl-dev** [py-pdf/fpdf2#409] [Q&A]
PDF/X-1a is the Adobe standard / version required by KDP (Kindle Direct Publishing).

Does fpdf2 comply with PDF/X-1a?

**Comment by Lucas-C** [comment]
Hi and thank you for asking this question.

I'm discovering this standard just now. Quoting the [Wikipedia page on PDF/X](https://en.wikipedia.org/wiki/PDF/X):

> PDF/X is a subset of the [PDF](https://en.wikipedia.org/wiki/Portable_Document_Format) [ISO](https://en.wikipedia.org/wiki/International_Organization_for_Standardization) standard. The purpose of PDF/X is to facilitate graphics exchange, and it therefore has a series of printing-related requirements which do not apply to standard PDF files. For example, in PDF/X-1a all fonts need to be embedded and all images need to be [CMYK](https://en.wikipedia.org/wiki/CMYK) or [spot colors](https://en.wikipedia.org/wiki/Spot_color).

As `fpdf2` does not require all images to be CMYK (it usually inserts images with a `/ColorSpace /DeviceRGB`), I can quickly answer you: **no, fpdf2 is currently not PDF/X-1a compliant**.

I also checked a PDF document generated with `fpdf2` with [](https://bfo.com/blog/2017/11/08/verify_pdfa_online/), with `PDF/X-1a` checkbox ticked, and this was the result:
<img width="527" alt="2022-05-04 21_58_19-Verify PDFA online" src="https://user-images.githubusercontent.com/925560/166816183-1759ef7e-8241-4ea6-ba0b-902d816e742a.png">

Contributions would of course be welcome to allow `fpdf2` to produce PDF/X-1a compliant documents.

For reference, I found some websites that list the criteria contained in the `PDF/X-1a:2003` spec:
* https://www.prepressure.com/pdf/basics/pdfx-1a
* https://www.normaprint.fr/blog/pdfx-1a2003/ (French)

Also, an issue somehow related: https://github.com/PyFPDF/fpdf2/issues/262

**Comment by dgl-dev** [comment]
Thanks, 
I'm going to try OpenOffice => PDF => Calibre => epub2 for KDP Books
I'm told that KDP is all happy with OpenOffice => PDF 
So if I can copy pages in OpenOffice and save as PDF, I'm good to go

If not of if it's funky, I'll get back about helping with the standards compliance

Les Dunaway
770-490-7546 <http://voice.google.com/calls?a=nc,%2B17704907546>
